lib/Behavior/Mapping/LocaleBehavior.php

```diff
@@ -29,4 +29,18 @@ public function getLocale();
      * @param string $locale
      */
     public function setLocale($locale);
+
+    /**
+     * Return the documents original locale.
+     *
+     * @return string
+     */
+    public function getOriginalLocale();
+
+    /**
+     * Sets the document original locale.
+     *
+     * @param string $locale
+     */
+    public function setOriginalLocale($locale);
```

Comment:
also for ghost pages this locale will be used. this will also be different to locale for snippets if there ghost content is loaded.